ghc.nix

```diff
@@ -20,7 +20,7 @@
 , withDwarf ? nixpkgs.stdenv.isLinux  # enable libdw unwinding support
 , withNuma ? nixpkgs.stdenv.isLinux
 , withDtrace ? nixpkgs.stdenv.isLinux
-, withGrind ? true
+, withGrind ? !(nixpkgs.valgrind.meta.broken or false)
```

Comment:
Hey @Gabriella439 ,

Thanks a lot for your contribution! In principle this change looks reasonable to me.
But, I have to admit I don't get why the `or false` is needed? :thinking:

BTW, @bgamari do you remember why Valgrind is added by default? Do most GHC devs use it?

Reply:
According to https://nixos.org/manual/nix/stable/language/operators.html, it means "if `broken` field is present, select its value, if absense then default to `false`". It's not a boolean `or` operation, but rather a syntactic sugar to provide fallback value when selecting an attribute.

Reply:
Yeah, it just prevents Nix from failing with an evaluation error if the `broken` field is not set